In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

# **Задача №1**

Не соблюдается условие нормальности | Две независимые выборки

**x1:** 380, 420, 290

**y1:** 140, 360, 200, 900

Сделайте вывод по результатам, полученным с помощью функции.

**Решение:**

Используем непараметрические тест + **критерий Манна- Уитни**.

In [2]:
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])

In [3]:
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

**Ответ:**

- pvalue=0.62 > а = 0.05

Принимаем нулевую гипотезу - статистически важных различий нет.


# **Задача №2**

Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. 

Есть ли статистически значимые различия?

**1е измерение до приема препарата:** 150, 160, 165, 145, 155

**2е измерение через 10 минут:** 140, 155, 150, 130, 135

**3е измерение через 30 минут:** 130, 130, 120, 130, 125

**Решение:**
Так как про нормальность нам неизвестно, используем непараметрические тест.

Множественное сравнение зависимых данных --> **критерий Фридмана**

In [5]:
t0 = np.array ([150, 160, 165, 145, 155])
t10 = np.array ([140, 155, 150, 130, 135])
t30 = np.array ([130, 130, 120, 130, 125])

stats.friedmanchisquare(t0, t10, t30)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

**Ответ:**

- pvalue=0.008 **<** а = 0.05

Принимаем альтернативную гипотезу: есть статистически значимые различия.

# **Задача №3**

Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

**1е измерение до приема препарата:** 150, 160, 165, 145, 155

**2е измерение через 10 минут:** 140, 155, 150, 130, 135

Есть ли статистически значимые различия?

**Решение:**

Так как о нормальности нам ничего неизвестно, используем непараметрические тест.

Две зависимые выборки --> используем **критерий Уилкоксона.**

In [6]:
t0 = np.array ([150, 160, 165, 145, 155])
t10 = np.array ([140, 155, 150, 130, 135])

stats.wilcoxon(t0, t10)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

**Ответ:**

- pvalue=0.0625 **>** а = 0.05

Принимаем нулевую гипотезу - статистически важных различий нет.

# **Задача №4**

Даны 3 группы учеников плавания. Время на дистанцию 50 м составляют

**Первая группа:** 56, 60, 62, 55, 71, 67, 59, 58, 64, 67

**Вторая группа:** 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

**Третья группа:** 57, 67, 49, 48, 47, 55, 66, 51, 54

Есть ли различия между группами?

**Решение:**

Так как о нормальности нам ничего неизвестно, используем непараметрические тест.

Множественное сравнение независимых данных --> используем **критерий Крускала – Уоллиса(Краскела – Уоллиса).**

In [7]:
gr_1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr_2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr_3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(gr_1, gr_2, gr_3)


KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

**Ответ:**

- pvalue=0.065 **>** а = 0.05

Принимаем нулевую гипотезу - статистически важных различий нет.

# **Задача №5**

Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

**Решение:**

Нормальное распределение, используем параметрические тест + **t-критерий (критерий Стьюдента)**, т.к. 𝞼 генеральной совокупности неизвестна.

$$t = \frac{\overline{x} - μ} {\sqrt{\frac{D(X)}{n}}}$$

$$ H_0: μ_1 = μ_0 = 2.5 $$

$$ H_1: μ_1 \neq μ _0$$

α(альфа) = 0,05(5%) - по условию


In [9]:
nums = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
disp = np.var(nums, ddof=1)
aver = np.mean(nums)
μ = 2.5
t = (aver - μ) /np.sqrt(disp/len(nums))
t

0.5630613661802959

Табличное значение t-критерия для двусторонего теста (а = 0,025)

In [10]:
t_t = stats.t.ppf(0.975, len(nums)-1)
print(f'Табличное значение t-критерия: {t_t: .3f}')


Табличное значение t-критерия:  2.262


**Ответ:**

Принимаем **нулевую** гипотезу с уровенем статистической значимости 5%, что партия изготавливается со средним арифметическим 2,5 см.